### PR TITLE
Fixes #29085 - Fix Style/SafeNavigation cop

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: modules/dns_dnscmd/dns_dnscmd_main.rb [hunk 1]
```diff
@@ -50,7 +50,7 @@ def execute cmd, msg=nil, error_only=false
       rescue Timeout::Error
         raise Proxy::Dns::Error.new("dnscmd did not respond within #{tsecs} seconds")
       ensure
-        std_in.close  unless std_in.nil?
+        std_in&.close
         std_out.close unless std_in.nil?
         std_err.close unless std_in.nil?
```
Comment (mmoll) [inline modules/dns_dnscmd/dns_dnscmd_main.rb:55]: While I have no idea about this code, I guess these two should get adjusted.

File: lib/proxy/log_buffer/decorator.rb [hunk 2]
```diff
@@ -50,7 +50,7 @@ def add(severity, message = nil, progname = nil, backtrace = nil)
       end
       info("Logging file reopened via USR1 signal") if reopened
       # exceptions are also sent to structured log if available
-      self.exception("Error details", backtrace) if backtrace && backtrace.is_a?(Exception)
+      self.exception("Error details", backtrace) if backtrace&.is_a?(Exception)
```
Comment (ekohl) [inline lib/proxy/log_buffer/decorator.rb:53]: Is this actually needed? Doesn't `is_a?` always work, even on nil?

File: modules/dns_dnscmd/dns_dnscmd_main.rb (hunk 1 above)
Comment (ekohl) [reply modules/dns_dnscmd/dns_dnscmd_main.rb:55]: This looks like a bug indeed

**Review by ekohl — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

File: lib/proxy/log_buffer/decorator.rb (hunk 2 above)
Comment (lzap) [reply lib/proxy/log_buffer/decorator.rb:53]: Correct.

File: modules/dns_dnscmd/dns_dnscmd_main.rb (hunk 1 above)
Comment (lzap) [reply modules/dns_dnscmd/dns_dnscmd_main.rb:55]: Yeah definitely, just adjust everything to `&.` please.

**Comment by lzap** [comment]
This still needs a fix, let me know if you are planning a PR:

```
        std_in&.close
        std_out.close unless std_in.nil?
        std_err.close unless std_in.nil?
```

File: modules/dns_dnscmd/dns_dnscmd_main.rb (hunk 1 above)
Comment (ekohl) [reply modules/dns_dnscmd/dns_dnscmd_main.rb:55]: Note that it's always checking `std_in` here so the cop can't detect it's the same.

**Comment by mmoll** [comment]
https://github.com/theforeman/smart-proxy/pull/703

**Pull request merged**